include/deal.II/grid/tria_objects_orientations.h

```diff
@@ -0,0 +1,317 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2022 - 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_tria_objects_orientations_h
+#define dealii_tria_objects_orientations_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/memory_consumption.h>
+#include <deal.II/base/utilities.h>
+
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace internal
+{
+  namespace TriangulationImplementation
+  {
+    /**
+     * Class storing 3D orientation information for various objects in a
+     * Triangulation.
+     *
+     * In deal.II, we express the orientation of an object with three
+     * booleans: orientation, rotate, and flip. The default values for these
```

Comment:
We should merge `rotate` and `flip` at some time, since they only indicate the number of rotation steps by 60 or 90 degree.

Reply:
can you open an issue with what you have in mind here? We still need two bits to store this information.